### DISCUSSION TITLE
How to get component name

### DISCUSSION THREAD

**Discussion by uissoJoseph** [livewire/livewire#6330] [2. Show and tell]
from v2 i was able to get component name like this

`...
 PlatformForm::getName();
//>> forms.platform-form
`
but in v3 got error say
Non-static method Livewire\Component::getName() cannot be called statically

**Comment by stefanschindler** [comment]
@uissoJoseph did you find any solution to that? We are also having the same problem after upgrading to v3

**Reply by nerocil** [reply]
I end up create custom class to handle this issue 
Its temporary  ugly solution 

```
class ComponentName
{
    public static function getName(string $className): string
    {
        if (! Str::startsWith($className, 'App\Livewire\\')) {
            throw new \InvalidArgumentException('Class name must be Livewire Component. Passed class is '.$className);
        }

        $name = Str::replace('App\Livewire\\', '', $className);
        $name = Str::replace('\\', '.', $name);

        $componentName = Str::afterLast($name, '.');
        $componentName = Str::snake($componentName, '-');

        $name = Str::beforeLast($name, '.');
        $name = Str::lower($name);
        $name = Str::snake($name, '-');

        return "{$name}.{$componentName}";
    }
}
```
Usage 
` $componentName = ComponentName::getName(EmployeeForm::class);`

please share if you have better solution.

**Reply by stefanschindler** [reply]
@nerocil 

Thanks! we ended up doing something similar like you. I don't really like this solution to be honest, but couldn't find anything better yet.

**Comment by PierreLebedel** [comment]
Hi! @uissoJoseph, @stefanschindler, @nerocil 

To get component name by class (or get class by component name), you can use the Livewire Component Registry which is accessible in Laravel App Container : `$componentName = app(Livewire\Mechanisms\ComponentRegistry::class)->getName(PlatformForm::class);`

I personnaly use this to open Modal component from its Trait : `$modalName = app(ComponentRegistry::class)->getName(self::class);`

**Reply by nerocil** [reply]
Thank you @PierreLebedel ,   this is better

**Reply by stefanschindler** [reply]
Thanks @PierreLebedel , i also like that solution better

**Comment by doonfrs** [comment]
LiveWire uses Str::kebab function internally to convert from PascalCase to the Kebab case, it is safer to use it, but you may remove the namespace first

```
$classParts = explode('\\', $classWithNamespace);
$className = end($classParts);
```

`
$widgetName = Str::kebab($widgetName);
`
